src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -485,7 +485,8 @@ void CodeGen::genHWIntrinsic_R_RM(
                         // Until we improve the handling of addressing modes in the emitter, we'll create a
                         // temporary GT_IND to generate code with.
                         GenTreeIndir load = indirForm(rmOp->TypeGet(), addr);
```

Comment:
`GenTreeIndir load` can't be declared in the higher scope because it is a value type and there is no public empty constructor available.

Reply:
An empty constructor is defined under `DEBUGGABLE_GENTREE` here: https://github.com/dotnet/coreclr/blob/master/src/jit/gentree.h#L5023; but I figured it would be less desirable to expose that than to just call the `emitIns` method in the correct scope.

Reply:
> GenTreeIndir load can't be declared in the higher scope because it is a value type and there is no public empty constructor available.

I don't understand, what's stopping you from simply moving that declaration before the `if`?!

Reply:
👍, not sure why that didn't come to mind. It must be too early for me still 😄

Reply:
Well, it's probably not the most intuitive thing to do, creating an object that's not always needed. But the overall approach of pulling IR nodes out of thin air is basically a hack so why not :smile: